pkg/condition/condition.go

```diff
@@ -75,6 +75,23 @@ func (c *EvaluableCondition) Compile() error {
 	return nil
 }
 
+func (c *EvaluableCondition) CastContextToTypedParameters(context map[string]any) (map[string]any, error) {
```

Comment:
Mind adding a godoc comment here to note that this converts the provided context to typed condition parameters and returns an error if any additional context fields are provided that are not defined by the evaluable condition.